src/device-api/middleware.ts

```diff
@@ -0,0 +1,141 @@
+import * as morgan from 'morgan';
+import * as _ from 'lodash';
+
+import * as config from '../config';
+import * as apiKeys from '../lib/api-keys';
+import log from '../lib/supervisor-console';
+import { UpdatesLockedError } from '../lib/errors';
+
+import type { Request, Response, NextFunction } from 'express';
+import type { AuthorizedRequestHandler } from './types';
+
+/**
+ * Request logger
+ */
+export const apiLogger = morgan(
```

Comment:
Maybe we can rename this middleware to `logging`? As we have `middleware.auth`, then we would have `middleware.logging`